src/contrib/tests/multihost-tests.rst

```diff
@@ -0,0 +1,142 @@
+.. _multihost-tests:
+
+===============
+Multihost tests
+===============
+
+SSSD multihost tests are the closest our tests get to running SSSD in the real world. The multihost tests utilize a VM the tests are ran at, so no part of the setup is faked. This is also the test's biggest advantage, as long as you can prepare the test environment, the tests can then be used to test even Active Directory or FreeIPA integration. Also, unlike the cwrap tests or the unit tests, the multihost tests are typically good enough for distribution QE teams, so the multihost tests allow a collaboration between the team that typically just develops SSSD and the team that tests it.
+
+The disadvantage of the tests is that setting up the environment can be complex and the development loop (the time between modifying test, modifying the SSSD sources, deploying them to the test environment and running the tests) is much longer than with the cwrap based tests.
+
+Running multihost tests
+-----------------------
+
+First, the infrastructure does not yet concern itself with provisioning at all. You need to set up a VM to run the tests on yourself. As an example, we'll be running the tests on a Fedora 28 machine which we will create using the `vagrant <https://www.vagrantup.com>`_ tool. To make things at least a little more palatable, the tests are expected to clean up after themselves, so it's possible to reuse the same provisioned VM for multiple test runs.
```

Comment:
Can we replace it with an example of running multihost test on sssd-test-suite?

Reply:
You don't need to setup the root password now with sssd-test-suite. Also it is not currently possible to run other tests then `basic` (well it is possible but not straighforward, I still need to work on that). So only basic tests should be run now and you need to specify the directory so other tests are not picked up (it is currently not possible to run pytest from the root multihost dir).

Check this for guide: https://github.com/SSSD/sssd/blob/master/src/tests/multihost/README.md
But perhaps keep including the virtual environment (with a meaningful name?)